src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -1743,6 +1756,8 @@ def coefficients_monomials(self, order=None, sparse=True):
         A = Matrix(R, len(self), len(v), sparse=sparse)
         for x, poly in enumerate(self):
             for m in poly:
+                if isinstance(m, tuple):
+                    m = m[1]
                 try:
                     A[x, y[m]] = one
                 except KeyError:
```

Comment:
I'd rather use `isinstance(self.ring(), BooleanPolynomialRing_base)`, and perhaps (don't know) do the check outside of the loops.

Reply:
I've changed it up a bit to check the type of polynomial ring outside of the loops.